lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1957,6 +1968,13 @@ static bool ParseIRGenArgs(IRGenOptions &Opts, ArgList &Args,
                      : "-gdwarf_types");
   }
 
+  for (auto A : Args.getAllArgValues(options::OPT_file_prefix_map)) {
```

Comment:
Does order of these matter? Like do we want debug-prefix-map entries to take priority over these? If so this might need to go below the other 2 assignments so these entries are last? (maybe this shouldn't be relied on though)

Reply:
The order matters if one remapping is a prefix of another. Swift stores them in a vector, so the order matters, but for Clang, it's stored in an std::map which will sort using std::greater. On the other hand, Clang [puts](https://cs.github.com/llvm/llvm-project/blob/7726ad04e2633a843fb9743e5adfbb404446e657/clang/lib/Driver/ToolChains/Clang.cpp#L691) the `-ffile-prefix-map` arguments first.

Reply:
Hmm I think I might have read that code wrong, it looks like it might preserve the order of the arguments - we could do that as well.

Reply:
Done, mirrored the Clang style which should preserve the ordering.